### PR TITLE
[WIP] work on DICOM anonymisation

### PR DESCRIPTION
**Pull request by JacobMcAloney**
fix bug with copy, allow deletion of unknown tags, create glob method anonymisation

### CODE REVIEW

File: src/pymedphys/dicom/_level2/anonymise.py [hunk 1]
```diff
@@ -156,13 +168,9 @@ def anonymise_dicom_dataset(ds, replace_values=True, delete_private_tags=True,
...
-        tags_used = list(ds.keys())
-        non_private_tags_used = np.array([
-            tag for tag in tags_used if not tag.is_private
-        ])
-        are_tags_used_in_dict_copy = [
-            key in BaselineDicomDictionary.keys()
-            for key in non_private_tags_used]
+        (
+            non_private_tags_used, are_tags_used_in_dict_copy
+        ) = check_if_all_tags_are_in_baseline(ds)
```
Comment (Centrus007) [inline src/pymedphys/dicom/_level2/anonymise.py]: Yuck! `autopep8` generating ridiculous line overflow corrections again?

**Comment by SimonBiggs** [comment]
> How do you feel about attempting to write an automated test for this change?

I'll take that baton.

**Comment by SimonBiggs** [comment]
Close in favour of https://github.com/pymedphys/pymedphys/pull/132

**Comment by SimonBiggs** [comment]
Will be merged with @Centrus007 pull request.

**Pull request closed without merging**